### DISCUSSION TITLE
File not open

### DISCUSSION THREAD

**Discussion by ocsike** [anggrayudi/SimpleStorage#139] [Q&A] [answered]
Hi,
I have two activity Activity1 and Activity2 with using SimpleStorageHelper class. In Activity1 I pick file and strore filepath into a database. In Activity2 I read the filepath from datatbase and try to open with DocumentFileCompat.fromFullPath method, but it does not work, unless in Activity1 I call openfolderpicker too. What's wrong in my solution?

**Comment by anggrayudi** [comment, marked as answer]
Android 10 framework does not allow you to retain the file access once the app is killed, that's why `DocumentFileCompat.fromFullPath()` returned `null`. So, storing the file path into a database is not a good idea, unless you have the URI access granted for its parent folder. So, if you want to keep the file access in long-term purpose, you will need to request access to its parent folder first. It can be done via `SimpleStorage#requestStorageAccess()`.

Suppose that your file is in `/storage/emulated/0/Pictures/selfie.jpg`, then you need to request access for `/storage/emulated/0/Pictures`, so accessing `selfie.jpg` will be possible in the future. Example:
```kotlin
storageHelper.onStorageAccessGranted = { _, root ->
    Toast.makeText(
        this,
        getString(com.anggrayudi.storage.R.string.ss_selecting_root_path_success_without_open_folder_picker, root.getAbsolutePath(this)),
        Toast.LENGTH_SHORT
    ).show()
}
storageHelper.requestStorageAccess(initialPath = FileFullPath(this, "/storage/emulated/0/Pictures"))
```
After the user granted access to your request, then `DocumentFileCompat.fromFullPath()` will return a result:
```kotlin
val picture = DocumentFileCompat.fromFullPath(this, "/storage/emulated/0/Pictures/selfie.jpg")
```

**Comment by ocsike** [comment]
Ok, thanks.
